app/lib/methods/handleMediaDownload.ts

```diff
@@ -199,18 +221,23 @@ export function downloadMediaFile({
 		try {
 			const path = getFilePath({ type, mimeType, urlToCache: downloadUrl });
 			if (!path) {
-				reject();
-				return;
+				return reject();
 			}
 			downloadKey = mediaDownloadKey(downloadUrl);
 			downloadQueue[downloadKey] = FileSystem.createDownloadResumable(downloadUrl, path);
+			if (isImageBase64(downloadUrl)) {
+				// https://stackoverflow.com/a/63186487
+				downloadUrl = downloadUrl.replace(imageBase64RegExp, '');
+				await FileSystem.writeAsStringAsync(path, downloadUrl, { encoding: FileSystem.EncodingType.Base64 });
+				return resolve(path);
+			}
 			const result = await downloadQueue[downloadKey].downloadAsync();
 			if (result?.uri) {
 				return resolve(result.uri);
 			}
-			reject();
-		} catch {
-			reject();
+			return reject();
+		} catch (e) {
+			return reject();
 		} finally {
 			delete downloadQueue[downloadKey];
```

Comment:
I'm _pretty_ sure `finally` runs even with an early `return` like the one you added, but please test to confirm it's actually removing from queue.

Reply:
Yes, it is working. This `return` fixes that infinite loading when occurs any problem and the file keeps the state of downloading, instead back the state to `to download` or `downloaded`